src/Icons/src/DependencyInjection/UXIconsExtension.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Icons\DependencyInjection;
+
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Loader\PhpFileLoader;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\HttpKernel\DependencyInjection\ConfigurableExtension;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @internal
+ */
+final class UXIconsExtension extends ConfigurableExtension implements ConfigurationInterface
+{
+    public function getConfigTreeBuilder(): TreeBuilder
+    {
+        $builder = new TreeBuilder('ux_icons');
+        $rootNode = $builder->getRootNode();
+
+        $rootNode
+            ->children()
+                ->scalarNode('icon_dir')
+                    ->info('The local directory where icons are stored.')
+                    ->defaultValue('%kernel.project_dir%/templates/icons')
+                ->end()
+                ->scalarNode('twig_component_name')
+                    ->info('The name of the Twig component to use for rendering icons.')
+                    ->defaultValue('UX:Icon')
```

Comment:
We should not do this. 

If someone want to use a custom component, he shall create its own. But he won't benefit from all the optimisations we will do.

But making this configurable will be a hell to handle / maintain in the TwigComponent bundle, and will add even more "if cases" in places where they should not be handled.

Not speaking about all the unavoidable "why can't I create a <foo:bar component" questions that will create immediatly :|

Reply:
Ok, i realize you "just" wanted to allow `<twig:Foo:Bar />` here ... 

As i understood it, `<UX:Icon />` was something a user could chose.

So sorry... still think we should not let this configurable, but i'm less afraid :))

Reply:
I'm on the fence about between:

A) Keeping this
B) Jumping in and eating out own dog food by adding a proper way to "import" 3rd party components and leveraging it here. I just really hate having the YAML config to do this :p

Reply:
I've removed the ability to customize for now.